### PR TITLE
Specifications cleanup

### PR DESCRIPTION
**Pull request by simon-ess**
This does a lot of needed cleanup to the build system document. Still a WIP.

### CODE REVIEW

File: build-system/specifications.md [hunk 1]
```diff
@@ -0,0 +1,4606 @@
...
+linker options for any arch that does not have a `USR_LDFLAGS_<osclass>`
+definition
+
+#### Options for a target specific compile/link command.
+
+```
+<name>_INCLUDES += -I<name>
+```
+
+header file directories each prefixed by a "-I".
```
Comment (gabrielfedel) [inline build-system/specifications.md:1192]: "-I" -> \`-I\`

File: build-system/specifications.md [hunk 2]
```diff
@@ -0,0 +1,4606 @@
...
+
+```
+<name>_INCLUDES_<osclass> += -I<name>
+```
+
+os specific header file directories each prefixed by a "-I".
```
Comment (gabrielfedel) [inline build-system/specifications.md:1198]: Same here

File: build-system/specifications.md [hunk 3]
```diff
@@ -0,0 +1,4606 @@
...
+<prodname>_SRCS_DEFAULT += <name>
+```
+
+Source files will be used for named product for any arch that does not
+have a `<prodname>_SRCS_<osclass>` definition
+
+#### Specifying libraries to be linked when creating the product
+
+For each library name specified which is not a system library nor a
+library from EPICS base, a <name>_DIR definition must be present in
```
Comment (gabrielfedel) [inline build-system/specifications.md:2467]: \<name\>_DIR -> \`\<name\>_DIR\ `
(right now \<name\> is not appearing on rendered version)

File: build-system/specifications.md [hunk 4]
```diff
@@ -0,0 +1,4606 @@
...
+The preexisting manifest file will be used for the created jar file.
+
+```
+JAR_PACKAGES += <name>
+```
+
+Names of java packages to be installed and added to the created jar
+file.
+
+#### Example 2 1
```
Comment (gabrielfedel) [inline build-system/specifications.md:2986]: Maybe Example 2.1  or 2-1 ?

File: build-system/specifications.md [hunk 5]
```diff
@@ -0,0 +1,4606 @@
...
+    | installed                        | NO) (default is NO). On win32 if |
+    |                                  | STATIC_BUILD=YES then set        |
+    |                                  | SHARED_LIBRARIES=NO)             |
+    +==================================+==================================+
+    | INC                              | list of include files to install |
+    |                                  | into $(INSTALL_DIR)/include      |
+    +----------------------------------+----------------------------------+
+    | INC_<osclass>                    | os specific includes to          |
+    |                                  | installed under                  |
+    |                                  | $(INSTALL_DIR)/include/os/<osclass>|
```
Comment (gabrielfedel) [inline build-system/specifications.md:3925]: The table is "breaking" here, but maybe is not a big issue.

File: build-system/specifications.md [hunk 6]
```diff
@@ -0,0 +1,4606 @@
...
+    |                                  | $(INSTALL_DIR)/html/$(HTMLS_DIR) |
+    +----------------------------------+----------------------------------+
+    | HTMLS                            | hypertext files to be installed  |
+    |                                  | into the                         |
+    |                                  | $(INSTALL_DIR)/html/$(HTMLS_DIR) |
+    |                                  | directory                        |
+    +----------------------------------+----------------------------------+
+    | TEMPLATES_DIR                    | template directory to be created |
+    |                                  | as                               |
+    |                                  | $(INSTALL_DIR)/templates/$(TEMPLATE_DIR) |
```
Comment (gabrielfedel) [inline build-system/specifications.md:4025]: Breaks here also, but maybe not a big issue.

File: build-system/specifications.md [hunk 7]
```diff
@@ -0,0 +1,4606 @@
...
+subdirectories to be built.
+
+`RULES_EXPAND`
+
+Definitions and rules to use `expandVars.pl` to expand @VAR@ variables
+in a file.
+
+`RULES_FILE_TYPE`
+
+Definitions and rules to allow user created CONFIG* and RULES* files
```
Comment (gabrielfedel) [inline build-system/specifications.md:4318]: Add \` \` to CONFIG* and RULES*

File: build-system/specifications.md [hunk 8]
```diff
@@ -0,0 +1,4606 @@
...
+Makefile code to create target specific dependency lines for libraries
+and product targets.
+
+`RULES_TOP`
+
+Rules specific to a `<top>` level directory e.g. uninstall and tar. It
+also includes the `RULES_DIRS` file.
+
+Makefile Definitions to allow creation of `CONFIG_APP_INCLUDE` and
+installation of the CONFIG* files into the `$(INSTALL_LOCATION)`
```
Comment (gabrielfedel) [inline build-system/specifications.md:4333]: Add \` \` to CONFIG*

**Comment by gabrielfedel** [comment]
I'm also fine to merge this, and apply the fixes later, as it is a very big page.

File: build-system/specifications.md (hunk 4 above)
Comment (simon-ess) [reply build-system/specifications.md]: The example names are kind of random; I switched them all to Example 1 2 3 4 5 6.

File: build-system/specifications.md [hunk 9]
```diff
@@ -0,0 +1,4588 @@
...
+
+Janet Anderson is the author of this chapter.
+
+## Overview
+
+This chapter describes the EPICS build facility including directory
+structure, environment and system requirements, configuration files,
+Makefiles, and related build tools.
+
+### `<top>`Directory structure
```
[the remark this answers is not in the document]
Comment (simon-ess) [reply build-system/specifications.md]: We agreed that we will leave these as is for the time being.

**Comment by gabrielfedel** [comment]
LGTM!

**Pull request merged**